### PR TITLE
Added mechanism to query retained instance store

### PR DESCRIPTION
**Pull request by LachlanMcKee**
## Description
Added mechanism to query retained instance store

## Check list

- [x] I have updated `CHANGELOG.md` if required.
- [x] I have updated documentation if required.

### CODE REVIEW

**Comment by CherryPerry** [comment]
What is use case?

**Review by CherryPerry — APPROVED** [review]

File: libraries/core/src/main/kotlin/com/bumble/appyx/core/store/RetainedInstanceStoreImpl.kt [hunk 1]
```diff
@@ -11,6 +11,22 @@ internal class RetainedInstanceStoreImpl : RetainedInstanceStore {
             .getOrPut(key) { ValueHolder(factory(), disposer) }
             .value as T
 
+    override fun isRetainedByStoreId(storeId: String, value: Any): Boolean =
+        map[storeId]
+            ?.let { valueHolderMap ->
```
Comment (KovalevAndrey) [inline libraries/core/src/main/kotlin/com/bumble/appyx/core/store/RetainedInstanceStoreImpl.kt]: [nit] could be a reusable function for both implementations
```
private fun MutableMap<String, ValueHolder<*>.hasValue(value: Any) {
    values.any { valueHolder ->
         valueHolder.value === value
    }
}
```

**Review by KovalevAndrey — APPROVED** [review]
LGTM, same question though :)

**Comment by LachlanMcKee** [comment]
Thanks! There is definitely a usecase when gradually migrating to the RetainedInstanceStore where you might want to optionally use the `disposables` plugin for a disposable object. 

The benefit of this new function is that when you don't want the retained object to be used within the `disposables` plugin, you could query whether it exists within the RetainedInstanceStore.

File: libraries/core/src/main/kotlin/com/bumble/appyx/core/store/RetainedInstanceStoreImpl.kt (hunk 1 above)
Comment (LachlanMcKee) [reply libraries/core/src/main/kotlin/com/bumble/appyx/core/store/RetainedInstanceStoreImpl.kt]: Good suggestion. I have applied it with a couple of tweaks

**Pull request merged**